diff -Naur ./CMakeLists.txt ./CMakeLists.txt
--- ./CMakeLists.txt	2020-02-17 18:31:08.595943959 -0500
+++ ./CMakeLists.txt	2020-02-17 18:34:30.743809490 -0500
@@ -1,6 +1,10 @@
 cmake_minimum_required(VERSION 2.8)
 project(GF_ORB_SLAM2)
 
+if(POLICY CMP0057)
+cmake_policy(SET CMP0057 NEW)
+endif()
+
 #set(CMAKE_POSITION_INDEPENDENT_CODE ON)
 
 message("ROS version = " $ENV{ROS_DISTRO})
@@ -114,7 +118,7 @@
     
 # customize eigen dir
 # find_package(Eigen3 3.1.0 REQUIRED)
-set( EIGEN3_INCLUDE_DIR /opt/eigen33/include/eigen3/ )
+set( EIGEN3_INCLUDE_DIR /opt/eigen/include/eigen3/ )
 # set( EIGEN3_INCLUDE_DIR /usr/include/eigen3 )
 MESSAGE("EIGEN include dir: " ${EIGEN3_INCLUDE_DIR})
 # MESSAGE("EIGEN libs: " ${EIGEN3_LIBS})
diff -Naur ./Thirdparty/g2o/CMakeLists.txt ./Thirdparty/g2o/CMakeLists.txt
--- ./Thirdparty/g2o/CMakeLists.txt	2020-02-17 18:31:08.959943720 -0500
+++ ./Thirdparty/g2o/CMakeLists.txt	2020-02-17 18:34:46.331799020 -0500
@@ -83,7 +83,7 @@
 
 # Find Eigen3
 # SET( G2O_EIGEN3_INCLUDE /usr/include/eigen3 )
-SET( G2O_EIGEN3_INCLUDE /opt/eigen33/include/eigen3/ )
+SET( G2O_EIGEN3_INCLUDE /opt/eigen/include/eigen3/ )
 SET(EIGEN3_INCLUDE_DIR ${G2O_EIGEN3_INCLUDE})
 FIND_PACKAGE(Eigen3 3.3.0 REQUIRED)
 IF(EIGEN3_FOUND)
